[doc] Fix markdownlint error for gpu-vulkan.md

hanhanW opened the pull request.

hanhanW committed:
[doc] Fix markdownlint for gpu-vulkan.md

antiagainst reviewed (approved):
Oops sorry. So the formating is not a blocking check before landing.. Thanks for fixing this!

ScottTodd commented on docs/website/docs/guides/deployment-configurations/gpu-vulkan.md:
```diff
@@ -193,22 +193,22 @@ iree-compile \
     to compile towards a specific GPU architecture.
-    
+
     We don't support the full spectrum here[^1]; the following table summarizes the
```
The lint error here can either be fixed with
```diff
+<!-- markdownlint-disable-next-line -->
[^1]: It's also impossible to capture all details of a Vulkan implementation
with a target triple, given the allowed variances on extensions, properties,
limits, etc. So the target triple is just an approximation for usage.
```

or with this patch: https://github.com/ScottTodd/iree/commit/697ef3cf526d72be58a77681242a3e770534b0ea

That patch uses an [annotation](https://squidfunk.github.io/mkdocs-material/reference/annotations/) instead of a [footnote](https://squidfunk.github.io/mkdocs-material/reference/footnotes/): 
![image](https://github.com/openxla/iree/assets/4010439/30627ee1-f6bc-40e4-9250-8409360b8ecb)

I don't have a strong preference for annotation vs footnote here.

ScottTodd replied:
Sent https://github.com/hanhanW/iree/pull/1 if you want to pull my patch onto this PR (I could also push to this PR directly, but figured I'd give some space to discuss the options)

hanhanW replied:
The annotation looks more straightforward to me. I merged the commit into my branch, thanks for the quick fix!

ScottTodd committed:
Switch from a footnote to an annotation to appease lint check. (#1)

The `MD053/link-image-reference-definitions` markdownlint rule does not look within (non-standard) admonitions.
